File: LibraryService.java
package com.library.management;

import java.util.*;

public class LibraryService {
    private List<Book> books = new ArrayList<>();
    private List<Member> members = new ArrayList<>();
    private List<Loan> loans = new ArrayList<>();
    
    public String borrowBook(String memberId, String bookId) {
        try {
            // Find member
            Member member = null;
            for (Member m : members) {
                if (m.getId().equals(memberId)) {
                    member = m;
                    break;
                }
            }
            
            if (member == null) {
                return "Member not found";
            }
            
            // Find book
            Book book = null;
            for (Book b : books) {
                if (b.getId().equals(bookId)) {
                    book = b;
                    break;
                }
            }
            
            if (book == null) {
                return "Book not found";
            }
            
            // Check if book is available
            if (!book.isAvailable()) {
                return "Book is not available";
            }
            
            // Create loan
            String loanId = UUID.randomUUID().toString();
            Loan loan = new Loan(loanId, memberId, bookId, new Date());
            loans.add(loan);
            
            // Update book status
            book.setAvailable(false);
            
            return "Book borrowed successfully. Loan ID: " + loanId;
            
        } catch (Exception e) {
            return "Error: " + e.getMessage();
        }
    }
    
    public String returnBook(String loanId) {
        try {
            // Find loan
            Loan loan = null;
            for (Loan l : loans) {
                if (l.getId().equals(loanId)) {
                    loan = l;
                    break;
                }
            }
            
            if (loan == null) {
                return "Loan not found";
            }
            
            // Find book
            Book book = null;
            for (Book b : books) {
                if (b.getId().equals(loan.getBookId())) {
                    book = b;
                    break;
                }
            }
            
            if (book != null) {
                book.setAvailable(true);
            }
            
            // Remove loan
            loans.remove(loan);
            
            return "Book returned successfully";
            
        } catch (Exception e) {
            return "Error: " + e.getMessage();
        }
    }
    
    public List<Book> getAvailableBooks() {
        List<Book> availableBooks = new ArrayList<>();
        for (Book book : books) {
            if (book.isAvailable()) {
                availableBooks.add(book);
            }
        }
        return availableBooks;
    }
    
    public List<Loan> getMemberLoans(String memberId) {
        List<Loan> memberLoans = new ArrayList<>();
        for (Loan loan : loans) {
            if (loan.getMemberId().equals(memberId)) {
                memberLoans.add(loan);
            }
        }
        return memberLoans;
    }
    
    public void addBook(Book book) {
        books.add(book);
    }
    
    public void addMember(Member member) {
        members.add(member);
    }
    
    public double calculateFine(String loanId) {
        Loan loan = null;
        for (Loan l : loans) {
            if (l.getId().equals(loanId)) {
                loan = l;
                break;
            }
        }
        
        if (loan == null) {
            return 0.0;
        }
        
        // Calculate days overdue
        long daysOverdue = (new Date().getTime() - loan.getBorrowDate().getTime()) / (1000 * 60 * 60 * 24);
        
        if (daysOverdue > 14) {
            return (daysOverdue - 14) * 0.50; // $0.50 per day after 14 days
        }
        
        return 0.0;
    }
}

File: python_library/library_service.py
import uuid
from datetime import datetime, timedelta
from typing import List, Optional

from .book import Book
from .member import Member
from .loan import Loan


class LibraryService:
    def __init__(self):
        self.books: List[Book] = []
        self.members: List[Member] = []
        self.loans: List[Loan] = []
    
    def borrow_book(self, member_id: str, book_id: str) -> str:
        try:
            # Find member
            member = None
            for m in self.members:
                if m.get_id() == member_id:
                    member = m
                    break
            
            if member is None:
                return "Member not found"
            
            # Find book
            book = None
            for b in self.books:
                if b.get_id() == book_id:
                    book = b
                    break
            
            if book is None:
                return "Book not found"
            
            # Check if book is available
            if not book.is_available():
                return "Book is not available"
            
            # Create loan
            loan_id = str(uuid.uuid4())
            loan = Loan(loan_id, member_id, book_id, datetime.now())
            self.loans.append(loan)
            
            # Update book status
            book.set_available(False)
            
            return f"Book borrowed successfully. Loan ID: {loan_id}"
            
        except Exception as e:
            return f"Error: {str(e)}"
    
    def return_book(self, loan_id: str) -> str:
        try:
            # Find loan
            loan = None
            for l in self.loans:
                if l.get_id() == loan_id:
                    loan = l
                    break
            
            if loan is None:
                return "Loan not found"
            
            # Find book
            book = None
            for b in self.books:
                if b.get_id() == loan.get_book_id():
                    book = b
                    break
            
            if book is not None:
                book.set_available(True)
            
            # Remove loan
            self.loans.remove(loan)
            
            return "Book returned successfully"
            
        except Exception as e:
            return f"Error: {str(e)}"
    
    def get_available_books(self) -> List[Book]:
        available_books = []
        for book in self.books:
            if book.is_available():
                available_books.append(book)
        return available_books
    
    def get_member_loans(self, member_id: str) -> List[Loan]:
        member_loans = []
        for loan in self.loans:
            if loan.get_member_id() == member_id:
                member_loans.append(loan)
        return member_loans
    
    def add_book(self, book: Book) -> None:
        self.books.append(book)
    
    def add_member(self, member: Member) -> None:
        self.members.append(member)
    
    def calculate_fine(self, loan_id: str) -> float:
        loan = None
        for l in self.loans:
            if l.get_id() == loan_id:
                loan = l
                break
        
        if loan is None:
            return 0.0
        
        # Calculate days overdue
        days_overdue = (datetime.now() - loan.get_borrow_date()).days
        
        if days_overdue > 14:
            return (days_overdue - 14) * 0.50  # $0.50 per day after 14 days
        
        return 0.0
